lib/feeb/db/repo/manager.ex

```diff
@@ -139,44 +164,98 @@ defmodule Feeb.DB.Repo.Manager do
     {:noreply, Map.put(state, queue_key, new_queue)}
   end
 
-  defp fetch_or_create_connection(:write, state) do
+  def handle_info({:DOWN, ref, _, caller_pid, reason}, state) do
+    keys = [:write_1, :read_1, :read_2]
+
+    state
+    |> Map.take(keys)
+    |> Enum.find(fn {_, %{monitor_ref: r}} -> r == ref end)
+    |> case do
+      # We found an active connection referencing this now-dead process. We need to make sure the
+      # connection is properly released
+      {key, %{pid: repo_pid, busy?: true, caller_pid: ^caller_pid}} ->
+        # Sanity check: by definition, the caller process is dead
+        false = Process.alive?(caller_pid)
```

Comment:
>  race conditions may occur

I don't think a race condition is possible in this case, but I'm not 100% sure, which is why I want to have a "blow everything up" protection in case it *does* happen.

If after some time using this code I realize this is impossible, then I'll just remove this assertion. Furthermore, since we do a clean-up at the Repo itself, even if the caller is still alive and somehow still manages to performs requests, the request will fail at the Repo layer (because it had its state reset).